src/core/server/saved_objects/service/lib/search_dsl/query_params.ts

```diff
@@ -39,17 +39,21 @@ function getTypes(mappings: IndexMapping, type?: string | string[]) {
 }
 
 /**
- *  Get the field params based on the types and searchFields
+ *  Get the field params based on the types, searchFields, and rawSearchFields
  */
-function getFieldsForTypes(types: string[], searchFields?: string[]) {
-  if (!searchFields || !searchFields.length) {
+function getFieldsForTypes(
+  types: string[],
+  searchFields: string[] = [],
+  rawSearchFields: string[] = []
+) {
+  if (!searchFields.length && !rawSearchFields.length) {
```

Comment:
OK, good idea and I'll add the check in the meantime until we get a final answer.

Reply:
> I would at least rename `searchFields` / `rawSearchFields` to something a little more explicit. `searchFields` -> `attributeSearchFields`? `typeSearchFields`?

Renaming this will touch a ton of other plugins, as discussed offline we'll leave this as-is (can always be renamed in the future if we decide to).

> `rawSearchFields` -> `rootSearchFields`?

Sounds good, I'll update this.